### ISSUE TITLE
JavadocTagContinuationIndentation Ignore indentation check when line breaks after the tag.

### ISSUE THREAD

**Issue by Anmol202005** [checkstyle/checkstyle#16656]
I have read check documentation: https://checkstyle.org/checks/javadoc/javadoctagcontinuationindentation.html#JavadocTagContinuationIndentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

follow up of https://github.com/checkstyle/checkstyle/issues/16385

Mentioned at https://github.com/checkstyle/checkstyle/pull/16518#discussion_r2008789048

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd"><module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocTagContinuationIndentation"/>
  </module>
</module>
```
```
/var/tmp $ cat Test.java
/**
 *
 * @see <a href="https://checkstyle.org/checks/javadoc">
 *   JavadocTagContinuationIndentation: Checks the indentation.</a> // violation raised
 *
 * @see
 *   <a href="https://checkstyle.org/checks/javadoc"> // false negative
 *   JavadocTagContinuationIndentation: Checks the indentation.</a>  // violation raised
 */
class Test {}
```
```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.22.0-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:6: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
[ERROR] /var/tmp/Test.java:10: Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]
Audit done.
Checkstyle ends with 2 errors.

```
When the line breaks immediately after the `javadoc_block tag` indentation rules should still be followed.
therefore, a violation is expected at line 9.
The implementation should first visit the JAVADOC_TAG, then traverse its children, such as REFERENCE, DESCRIPTION, or HTML_ELEMENT.

---

**Comment by Zopsss** [comment]
> The implementation should first visit the JAVADOC_TAG, then traverse its children, such as REFERENCE, DESCRIPTION, or HTML_ELEMENT.

There should be another issue opened for this, explaining why we're changing the implementation, add link of our conversation too for the context.

**Comment by romani** [comment]
> `avadoc"> // false negative`

Let's hold on on new issue.
Let's understand this for now. Please share exactly what is expected. I ses that there should be 2 violations, so all works well.

**Comment by romani** [comment]
Non of indentation is 4. So all should be violation. Please double check that CLI output is actual. Something is wrong.

**Comment by Anmol202005** [comment]
> Non of indentation is 4. So all should be violation. Please double check that CLI output is actual. Something is wrong.

i guess the next line after the line break in a `javadoc _block tag` should have an offset indentation of 4 
here in the mentioned case 
```java
 /**
 *
 * @see <a href="https://checkstyle.org/checks/javadoc">
 *   JavadocTagContinuationIndentation: Checks the indentation.</a> // violation raised
 *
 * @see
 *   <a href="https://checkstyle.org/checks/javadoc"> // false negative
 *   JavadocTagContinuationIndentation: Checks the indentation.</a>  // violation raised
 */
class Test {}
```
all the lines after the line break in a `javadoc _block tag` are flagged except the false negative .

am i missing anything?

**Comment by romani** [comment]
Is this issue about a fact that not all lines are flagged with violation ?

>var/tmp/Test.java:10

Line 10 is class, I am confused, please fix CLI output.

**Comment by Anmol202005** [comment]
>
> Line 10 is class, I am confused, please fix CLI output.

ya got it, i removed the package name from the top that's causing this ambiguity .

**Comment by Anmol202005** [comment]
@romani is it good now ?

**Comment by romani** [comment]
Now it is good and clear. Thanks a lot

**Comment by romani** [comment]
@Zopsss , issues usually explains change in behavior, this is what most users care about.

Details of implementation we change at any point of time,  summary of it will be preserved in PR description.

**Comment by Anmol202005** [comment]
@Brijeshthummar02 Yes it's open.

**Comment by Zopsss** [comment]
> Line continuation have incorrect indentation level, expected level should be 4. [JavadocTagContinuationIndentation]

We should also include the incorrect indentation level number in violation message, something like: 


> Line continuation have incorrect indentation level 2, expected level should be 4.

@romani @mahfouz72 @Anmol202005 what do you think?

**Comment by romani** [comment]
@Zopsss , better to discuss it in separate ticket, for now let keep as is - only expected.

this issue is good for Pull Request, everyone is welcome to do it.

**Comment by Atharv3221** [comment]
I am working on this issue

**Comment by Atharv3221** [comment]
```
    |   |       |       |--HTML_ELEMENT -> HTML_ELEMENT [7:7]                                             (doesn't include blank spaces before)
    |   |       |       |   |--HTML_TAG_START -> HTML_TAG_START [7:7]
    |   |       |       |   |   |--TAG_OPEN -> < [7:7]
    |   |       |       |   |   |--TAG_NAME -> a [7:8]
    |   |       |       |   |   |--HTML_ATTRIBUTES -> HTML_ATTRIBUTES [7:9]
    |   |       |       |   |   |   `--HTML_ATTRIBUTE -> HTML_ATTRIBUTE [7:9]
    |   |       |       |   |   |       |--TEXT ->   [7:9]
    |   |       |       |   |   |       |--TAG_ATTR_NAME -> href [7:10]
    |   |       |       |   |   |       |--EQUALS -> = [7:14]
    |   |       |       |   |   |       `--ATTRIBUTE_VALUE -> "https://checkstyle.org/checks/javadoc" [7:15]
    |   |       |       |   |   `--TAG_CLOSE -> > [7:54]
    |   |       |       |   |--HTML_CONTENT -> HTML_CONTENT [7:55] 
    |   |       |       |   |   |--TEXT ->  // false negative [7:55]
    |   |       |       |   |   |--NEWLINE -> \n [7:73]
    |   |       |       |   |   |--LEADING_ASTERISK ->  * [8:1]
    |   |       |       |   |   `--TEXT ->    JavadocTagContinuationIndentation: Checks the indentation. [8:3]   (does include blank spaces)

```
for 
```
 *   <a href="https://checkstyle.org/checks/javadoc"> // false negative
 *   JavadocTagContinuationIndentation: Checks the indentation.</a>  // violation raised
```
Also,
```
/**
 * Some class
 */
public class Test {
  /**
   * Some method
   */
  public void method() {}
}

```
Leading asterik column number is always reported 1

```
COMPILATION_UNIT -> COMPILATION_UNIT [4:1]
`--CLASS_DEF -> CLASS_DEF [4:1]
    |--MODIFIERS -> MODIFIERS [4:1]
    |   |--BLOCK_COMMENT_BEGIN -> /* [1:1]
    |   |   |--COMMENT_CONTENT -> *\n * Some class\n  [1:3]
    |   |   |   `--JAVADOC_CONTENT -> JAVADOC_CONTENT [1:4]
    |   |   |       |--NEWLINE -> \n [1:4]
    |   |   |       |--LEADING_ASTERISK ->  * [2:1]                                               <- Leading asterik
    |   |   |       |--TEXT ->  Some class [2:3]
    |   |   |       |--NEWLINE -> \n [2:14]
    |   |   |       `--TEXT ->   [3:1]
    |   |   `--BLOCK_COMMENT_END -> */ [3:2]
    |   `--LITERAL_PUBLIC -> public [4:1]
    |--LITERAL_CLASS -> class [4:8]
    |--IDENT -> Test [4:14]
    `--OBJBLOCK -> OBJBLOCK [4:19]
        |--LCURLY -> { [4:19]
        |--METHOD_DEF -> METHOD_DEF [8:3]
        |   |--MODIFIERS -> MODIFIERS [8:3]
        |   |   |--BLOCK_COMMENT_BEGIN -> /* [5:3]
        |   |   |   |--COMMENT_CONTENT -> *\n   * Some method\n    [5:5]
        |   |   |   |   `--JAVADOC_CONTENT -> JAVADOC_CONTENT [5:6]
        |   |   |   |       |--NEWLINE -> \n [5:6]
        |   |   |   |       |--LEADING_ASTERISK ->    * [6:1]                                   <- Leadinng asterik
        |   |   |   |       |--TEXT ->  Some method [6:5]
        |   |   |   |       |--NEWLINE -> \n [6:17]
        |   |   |   |       `--TEXT ->     [7:1] 
        |   |   |   `--BLOCK_COMMENT_END -> */ [7:4]
        |   |   `--LITERAL_PUBLIC -> public [8:3]
        |   |--TYPE -> TYPE [8:10]
        |   |   `--LITERAL_VOID -> void [8:10]
        |   |--IDENT -> method [8:15]
        |   |--LPAREN -> ( [8:21]
        |   |--PARAMETERS -> PARAMETERS [8:22]
        |   |--RPAREN -> ) [8:22]
        |   `--SLIST -> { [8:24]
        |       `--RCURLY -> } [8:25]
        `--RCURLY -> } [9:1]

```

**Comment by Atharv3221** [comment]
need to follow design of https://checkstyle.sourceforge.io/checks/javadoc/javadocleadingasteriskalign.html#JavadocLeadingAsteriskAlign

**Comment by ZaheerAhmadDev** [comment]
I am on it.

**Comment by romani** [comment]
fix is merged
